packages/mermaid/src/rendering-util/handle-markdown-text.js

```diff
@@ -65,30 +59,27 @@ export function markdownToLines(markdown) {
 }
 
 /**
- *
- * @param markdown
+ * @param {string} markdown markdown to convert to HTML
+ * @returns {string} HTML
  */
 export function markdownToHTML(markdown) {
-  const mdParse = SimpleMarkdown.defaultBlockParse;
-  const syntaxTree = mdParse(markdown);
+  const { children } = fromMarkdown(markdown);
```

Comment:
@remcohaszing reused `fromMarkdown` in this PR to reduce the scope. 

It'll still have issues with newlines, but we can resolve that in https://github.com/mermaid-js/mermaid/pull/4343 

The plan is to remove the react peer dependency in this release and properly fix the newline issues in a future release.

Reply:
FWIW I've created [`mdast-util-newline-to-break`](https://github.com/syntax-tree/mdast-util-newline-to-break).